r/R/dplyr-mutate.R

```diff
@@ -112,7 +113,15 @@ mutate.Dataset <- mutate.ArrowTabular <- mutate.RecordBatchReader <- mutate.arro
 
 transmute.arrow_dplyr_query <- function(.data, ...) {
   dots <- check_transmute_args(...)
-  dplyr::mutate(.data, !!!dots, .keep = "none")
+  has_null <- vapply(dots, quo_is_null, logical(1))
+  .data <- dplyr::mutate(.data, !!!dots, .keep = "none")
+  if (is_empty(dots) | any(has_null)) return(.data)
+
+  ## keeping with: https://github.com/tidyverse/dplyr/issues/6086
+  cur_exprs <- vapply(dots, as_label, character(1))
+  new_col_names <- names(dots)
+  transmute_order <- ifelse(nzchar(new_col_names), new_col_names, cur_exprs)
```

Comment:
I'm having a bit of trouble following what this line is doing here — what behavior are we trying to catch here?

Reply:
Very open to other approaches but the dplyr implementation of `transmute` return the columns in the order they are given in the function call which is now distinct from `mutate(..., .keep = "none")`.  So this was what i came up with to preserve that order. But there possibly could be a better way to capture order?

Reply:
`dplyr::coalesce` works here (and since this is a dplyr function, we can be pretty sure dplyr is already installed), but generally we try to shy away from using dplyr as a dependency. 

So I would say let's get the names from `cur_exprs` but then do the more base `transmute_order[is.na(names(...))] <- ...` assignment style. IMO it also is slightly more explicit about what we're doing (replacing anything that doesn't have a name with a name-like representation of itself)